File: kimdokyun/Python/boj_1339.py
alpaDic = {chr(i) : 0 for i in range(ord('A'), ord('Z')+1) }

N = int(input())
for _ in range(N):
    word = input()[::-1]
    digit = 1
    for alpa in word:
        alpaDic[alpa] += digit
        digit *= 10

result = 0
value = 9
for alpaV in sorted(alpaDic.values(),reverse=True):
    if alpaV == 0 : break
    result += alpaV * value
    value -= 1

print(result)

File: Closed/kimdokyun/C++/boj_1339.cpp
#include<stdio.h>
#include<string.h>
#include<algorithm>
using namespace std;

bool compare(int a, int b) {return b < a;}

int main() {
	int alpaDic[26] = {},N;
	scanf("%d",&N);
	for(int i=0;i<N;i++) {
		char word[9];
		int digit = 1;
		scanf("%s",word);
		for(int j=strlen(word)-1;j>=0;j--) {
			alpaDic[word[j]-'A'] += digit;
			digit *= 10;
		}
	}
	sort(alpaDic,alpaDic+26, compare);
	int result=0,value=9;
	for(int i=0;i<26 && value > 0 && alpaDic[i] > 0;i++) {
		result += alpaDic[i] * value--;
	}
	printf("%d",result);
}
